apps/root/src/hooks/useAllowance.ts

```diff
@@ -30,64 +30,29 @@ function useAllowance(
     error?: string;
   }>({ isLoading: false, result: dummyToken, error: undefined });
   const hasPendingTransactions = useHasPendingTransactions();
-  const prevFrom = usePrevious(from);
-  const prevPendingTrans = usePrevious(hasPendingTransactions);
-  const prevAccount = usePrevious(owner);
-  const currentNetwork = useSelectedNetwork();
-  const blockNumber = useBlockNumber(currentNetwork.chainId);
-  const prevBlockNumber = usePrevious(blockNumber);
   const prevResult = usePrevious(result, false, 'allowance');
-  const prevUsesYield = usePrevious(usesYield);
-  const prevVersion = usePrevious(version);
 
-  React.useEffect(() => {
+  const fetchAllowance = React.useCallback(() => {
     async function callPromise() {
       if (from) {
         try {
-          const promiseResult = await walletService.getAllowance(from, owner, usesYield, version);
+          const promiseResult = await walletService.getAllowance(from, owner as Address, usesYield, version);
           setState({ result: promiseResult, error: undefined, isLoading: false });
         } catch (e) {
           setState({ result: dummyToken, error: e as string, isLoading: false });
         }
       }
     }
 
-    if (
-      (!isLoading && !result && !error) ||
-      !isEqual(prevFrom, from) ||
-      !isEqual(owner, prevAccount) ||
-      !isEqual(prevPendingTrans, hasPendingTransactions) ||
-      !isEqual(prevUsesYield, usesYield) ||
-      !isEqual(prevVersion, version) ||
-      (blockNumber &&
-        prevBlockNumber &&
-        blockNumber !== -1 &&
-        prevBlockNumber !== -1 &&
-        !isEqual(prevBlockNumber, blockNumber))
-    ) {
-      setState({ isLoading: true, result: dummyToken, error: undefined });
+    setState({ isLoading: true, result: dummyToken, error: undefined });
 
-      // eslint-disable-next-line @typescript-eslint/no-floating-promises
-      callPromise();
-    }
-  }, [
-    from,
-    prevFrom,
-    isLoading,
-    result,
-    error,
-    usesYield,
-    prevUsesYield,
-    version,
-    prevVersion,
-    hasPendingTransactions,
-    prevAccount,
-    owner,
-    prevPendingTrans,
-    prevBlockNumber,
-    blockNumber,
-    walletService,
-  ]);
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    callPromise();
+  }, [from, isLoading, result, error, usesYield, version, hasPendingTransactions, owner, walletService]);
+
+  // When either of this values change we want to trigger this
+  React.useEffect(fetchAllowance, [owner, from?.address]);
```

Comment:
it was not triggering it :(

Reply:
Will create a ticket for it so we can advance with this

Reply:
useInteval should not retrigger the callback if it changes, it should only change its reference, since we want it to do it on an interval, so this is needed